### PR TITLE
Patch vulnerabilities

### PR DESCRIPTION
**Pull request by mntnorv**
Update Go and a few dependencies to patch vulnerabilities - CVE-2024-41110, CVE-2024-24790, CVE-2024-6257, CVE-2024-34156. All of them are false positives and kvisor is not affected.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -179,7 +179,7 @@ jobs:
       - name: Setup Go 1.21
         uses: actions/setup-go@v5
         with:
-          go-version: '1.21'
+          go-version: "1.21"
```
Comment (patrickpichler) [inline .github/workflows/build.yaml:182]: This should probably be 1.23 as well.

Comment (samuelvl) [reply .github/workflows/build.yaml:182]: We can also determine the go version to use in CI from the `go.mod` file and forget about inconsistencies in the future: https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Comment (patrickpichler) [reply .github/workflows/build.yaml:182]: TIL Thanks!

Comment (mntnorv) [reply .github/workflows/build.yaml]: Updated this place and made use of `go-version-file` everywhere

**Pull request merged**